### PR TITLE
plot_krona: debug incongruent abundances

### PR DESCRIPTION
**Pull request by erifa1**
Hi,
`unique()` function tends to remove taxas with real abundances and this causes inconcruencies between total sums of krona and `sample_sums(physeq)`. Removing taxas with 0 abundance solves this. 
This implies to have clean/non redundant taxonomy. 

Regards
Etienne

### CODE REVIEW

**Comment by cpauvert** [comment]
Hi Etienne, 
Thanks for your PR! I didn't actually thought people were using this^^
I look into your suggestion and go back to you soon!
Thanks,
Charlie

File: R/plot_functions.R [hunk 1]
```diff
@@ -245,10 +245,10 @@ plot_krona<-function(physeq,output,variable, trim=F){
...
   # and written to a file that would be processed by Krona.
   for( lvl in levels(df[,2])){
-    write.table(
-      unique(
-        df[ which( df[,2] == lvl), -2]
-        ),
+      tt = df[which(df[, 2] == lvl), -2]
+      tt2 = tt[which(tt[,1]!=0),]
+
+    write.table(tt2,
```
Comment (cpauvert) [inline R/plot_functions.R]: Well spotted. I think I did that initially on the toy dataset I used. But it does not make sense with real datasets.

Comment (cpauvert) [inline R/plot_functions.R]:
```suggestion
    write.table(
      df[which(df[, 2] == lvl & df[,1] != 0), -2]
```

Comment (cpauvert) [reply R/plot_functions.R]: Both statement can be combine and there will be no need for intermediate `tt` or `tt2` dataframes. If you're ok with that

File: R/plot_functions.R [hunk 2]
```diff
@@ -266,7 +266,7 @@ plot_krona<-function(physeq,output,variable, trim=F){
                "-o", output,
                sep = " "))
   # Run the browser to visualise the output.
-  browseURL(output)
+  # browseURL(output)
```
Comment (cpauvert) [inline R/plot_functions.R]:
```suggestion
  browseURL(output)
```
The function name is `plot_krona` so for consistency, I prefer to keep the plot by default.
But in a later commit, we could deactivate it with a `plot=FALSE` argument and have a default like this:
```r
if(plot) browseURL(output)
```

**Review by cpauvert — CHANGES_REQUESTED** [review]
Thanks @erifa1 for having spotted the buggy line. I just have minor request and the PR would be good to go.

File: R/plot_functions.R (hunk 1 above)
Comment (erifa1) [reply R/plot_functions.R]: Thanks for your reply, spotted this by using this function with big real dataset.

Comment (erifa1) [reply R/plot_functions.R]: No problem

File: R/plot_functions.R (hunk 2 above)
Comment (erifa1) [reply R/plot_functions.R]: No problem, this was actually just for me. i forgot this line in the commit.

**Review by erifa1 — COMMENTED** [review]
Thanks for considering my PR

**Review by cpauvert — APPROVED** [review]
Great!

**Pull request merged**